torch/_ops.py

```diff
@@ -309,7 +310,27 @@ def dispatch(self, dispatch_key, *args, **kwargs):
                 return handler(mode, *args, **kwargs)
 
         functionality_key = torch._C._to_functionality_key(dispatch_key)  # type: ignore[attr-defined]
-        if functionality_key in mode_stack_per_key():
+        if functionality_key == torch._C.DispatchKey.PreDispatch:
+            from torch.utils._python_dispatch import _pop_mode_temporarily
+
+            if (
+                _len_torch_dispatch_stack_pre_dispatch() > 0
+                and not torch._C._dispatch_tls_is_dispatch_key_excluded(
+                    DispatchKey.Python
```

Comment:
What do we we need the `is_dispatch_key_excluded(DispatchKey.Python)` check for?

Reply:
Likely not needed.